kernel_gateway/gatewayapp.py

```diff
@@ -312,19 +320,23 @@ def init_configurables(self):
                     self.prespawn_count, self.max_kernels)
                 )
 
-        if self.api == 'notebook-http':
-            self.kernel_pool = ManagedKernelPool(
-                self.prespawn_count,
-                self.kernel_manager
-            )
-        else:
-            self.kernel_pool = KernelPool(
-                self.prespawn_count,
-                self.kernel_manager
-            )
+        
+        api_module = self._load_api_module(self.api)
+        func = getattr(api_module, 'create_personality', None)
+        if func is None:
+            self.log.critical('ERROR: the personality factory function was not found in the API module {}'.format(self.api))
+            self.exit(1)
```

Comment:
Probably better to raise a RuntimeError here with the message you're logging as critical. `sys.exit(1)` is harsh when testing.